### PR TITLE
Upgrade to Crystal 0.35.0

### PR DESCRIPTION
**Pull request by straight-shoota**
These changes are required to make this shard work with Crystal 0.35.0

### CODE REVIEW

**Comment by da1nerd** [comment]
I literally just sat down to open a PR for this. :+1:

File: src/loader/string_encoder.cr [hunk 1]
```diff
@@ -39,5 +39,7 @@ class BakedFileSystem::StringEncoder < IO
         byte.to_s(16, @io, upcase: true)
       end
     end
+
+    slice.size.to_i64
```
Comment (carlhoerberg) [inline src/loader/string_encoder.cr:43]: that's not really correct, is it? a lot more bytes are actually written

Comment (straight-shoota) [reply src/loader/string_encoder.cr:43]: I think that's correct. At least it's the same in stdlib implementations.
The return value is supposed to be the number of bytes written from the input. Not how many bytes are passed on to the wrapped IO. That wouldn't have any meaning at all in the caller context.

**Comment by carlhoerberg** [comment]
Yes it would, the reason for returning the number of bytes actually written
is to make more efficient position accounting, that is not having to call
pos on the underlying IO.

File: src/loader/string_encoder.cr (hunk 1 above)
Comment (da1nerd) [reply src/loader/string_encoder.cr:43]: :exploding_head: wow I never knew that. Is that documented anywhere @straight-shoota ?

Comment (carlhoerberg) [reply src/loader/string_encoder.cr:43]: the intention of returing the number of bytes actually written is to simply position accounting. blindly returning slice.size doesn't help anyone, the consumer needs the bytes actually written.

Comment (straight-shoota) [reply src/loader/string_encoder.cr:43]: @neutrinog Unfortunately, not :(

Let's continue the discussion about intention in https://github.com/crystal-lang/crystal/pull/9233 or what ever comes from that.

**Comment by straight-shoota** [comment]
This PR is obsolete with the release of Crystal 0.35.1.

**Pull request closed without merging**